src/main.ts

```diff
@@ -35,24 +36,39 @@ import Settings = require("./settings");
 import { PowerShellLanguageId } from "./utils";
 import utils = require("./utils");
 
+// The most reliable way to get the name and version of the current extension.
+// tslint:disable-next-line: no-var-requires
+const PackageJSON: any = require("../../package.json");
+
 // NOTE: We will need to find a better way to deal with the required
 //       PS Editor Services version...
 const requiredEditorServicesVersion = "2.0.0";
 
+// the application insights key (also known as instrumentation key) used for telemetry.
+const AI_KEY: string = "AIF-d9b70cd4-b9f9-4d70-929b-a071c400b217";
+
 let logger: Logger;
 let sessionManager: SessionManager;
 let extensionFeatures: IFeature[] = [];
+let telemetryReporter: TelemetryReporter;
 
 const documentSelector: DocumentSelector = [
     { language: "powershell", scheme: "file" },
     { language: "powershell", scheme: "untitled" },
 ];
 
 export function activate(context: vscode.ExtensionContext): void {
+    // create telemetry reporter on extension activation
+    telemetryReporter = new TelemetryReporter(PackageJSON.name, PackageJSON.version, AI_KEY);
 
-    const version = getCurrentVersion(context);
+    // If both extensions are enabled, this will cause unexpected behavior since both register the same commands
+    if (PackageJSON.name.toLowerCase() === "powerShell-preview"
```

Comment:
You have a capital `S` in powerShell that will cause the toLowerCase comparison to always fail.

Reply:
Lol... 🤦‍♂️

Reply:
```suggestion
	    if (PackageJSON.name.toLowerCase() === "powershell-preview"
```